libraries/AP_HAL_Linux/I2CDevice.cpp

```diff
@@ -177,13 +177,13 @@ bool I2CDevice::transfer(const uint8_t *send, uint32_t send_len,
     i2c_data.msgs = msgs;
     i2c_data.nmsgs = nmsgs;
 
-    int r = -EINVAL;
+    int r;
     unsigned retries = _retries;
     do {
         r = ::ioctl(_bus.fd, I2C_RDWR, &i2c_data);
-    } while (r < 0 && retries-- > 0);
+    } while (r == -1 && retries-- > 0);
```

Comment:
Take a look at man 3 ioctl:

> Upon successful completion, ioctl() shall return a value other than −1 that depends upon the STREAMS device control function. Otherwise, it shall return −1 and set errno to indicate the error.

We aren't interested in the particular 'success' code. We only need to know if an error occured, and error case is only -1.